apps/sms/views/conversation/js/compose.js

```diff
@@ -4,10 +4,7 @@
 /*global Settings, Utils, Attachment, MozActivity, SMIL,
```

Comment:
> As far as I understood you wanted to pass entire view (that implements some predefined interface to access recipients in a unified way) as parameter to Compose constructor. I'm not really sure about the following things in this solution: Compose will know what recipients is and it will have access to entire view method/property set. But maybe I'm missing something.

Yeah, in the Java world you would take an interface as parameter so you wouldn't have access to the full object (unless you'd cast but this is bad practice). In JavaScript it's not easy to do the same thing so passing only the needed methods (possibly in an object to have an interface-like) is appealing.


I think `Compose.setupLock` would work fine in a splitted view app, but I don't like it in the main app architecture as this is really a changing state (as opposed to a constructor with 2 different instances where you have 1 non-changing state for each instance).

In the long-term we don't really care which one is  the best because we'll have split views for NewMessageView and ConversationView. So let's not spend too much time debating. Your solution would not be my solution of choice if we'd stayed unsplitted, but because we'll split, and it does not really introduce more state than what we have now, it's as good as what I have in mind, and likely easier to do as well.